lib/rules/no-restricted-imports.js

```diff
@@ -333,6 +398,25 @@ module.exports = {
 		 * @private
 		 */
 		function checkRestrictedPathAndReport(importSource, importNames, node) {
+			// Skip if this is a type-only import and it's allowed
+			if (
+				(node.type === "ImportDeclaration" ||
+					node.type === "TSImportEqualsDeclaration") &&
+				isTypeOnlyImport(node) &&
+				isAllowedTypeImportPath(importSource)
+			) {
+				return;
+			}
```

Comment:
This logic might not be correct when there are multiple entries for the same import source. For example:

```ts
/* eslint no-restricted-imports: ["error", {
    paths: [
        {
            name: "mod",
            importNames: ["foo"],
            allowTypeImports: true,
            message: "import 'foo' only as type"
        },
        {
            name: "mod",
            importNames: ["bar"],
            message: "don't import 'bar' at all"
        }
    ]
}]*/

import type { bar } from "mod"; // ok, but should be error?

```

Reply:
Hmm, seems to be an existing bug in[ @typescript-eslint/no-restricted-imports rule](https://typescript-eslint.io/play/#ts=5.8.2&fileType=.ts&code=PQKgBApgzgNglgOwC5gAJIJ4AdoGMBOcWSAtNPMsAgPYn7RKG5IQAmJcAtltfklAC4wAbQBEEfPl6iANGADeAKDAqwWAIZIAFoJHLVBpQeOqE6zhCGjO1VrP0mDXHnwBy56ELEAzatVEAujIOjirqMDDUAO4AKtgQAJLcvPxCjACuEMGhBhZQUOoA5pZgos4pYADkvtSVYNQIMBhg6lBgmDiiIQYAvtmORjkqZhZWNnb9Q%2BVuHrpiAEbq%2BIGTOXkFxVasDZUo0yiVi-h1mi0RXaE9IQGKPQEgwIqK%2B%2B3xCmBHYD1g3lKcpeNRABuIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQHYHsBaaFAF2gEsBjUxAE0IoFsAHfaU1DSfAM18gAacNkgABUgE8WKKpRalCKeBVykA9AWJlKNeo1btO6MAG1IiaNHZDMAXwC6IuyDtA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false), will fix.

Reply:
Fixed in https://github.com/eslint/eslint/pull/19562/commits/9521fb99949641da52c204a0c880980d657c2c7b